packages/drivers/routerlicious-driver/src/documentService.ts

```diff
@@ -200,15 +202,31 @@ export class DocumentService implements api.IDocumentService {
 							refreshToken,
 						}),
 					},
-					async () => {
-						const newOrdererToken = await this.tokenProvider.fetchOrdererToken(
-							this.tenantId,
-							this.documentId,
-							refreshToken,
-						);
-						this.ordererRestWrapper.setToken(newOrdererToken);
-						return newOrdererToken;
-					},
+					async () =>
+						this.tokenProvider
+							.fetchOrdererToken(this.tenantId, this.documentId, refreshToken)
+							.then(
+								(newOrdererToken) => {
+									this.ordererRestWrapper.setToken(newOrdererToken);
+									return newOrdererToken;
+								},
+								(error) => {
+									const rawCanRetry = error?.canRetry;
+									const tokenError = wrapError(
+										error,
+										(errorMessage) =>
+											new NetworkErrorBasic(
+												`The Host-provided token fetcher threw an error`,
+												OdspErrorType.fetchTokenError,
```

Comment:
You probably do not want to have any ODSP related dependecies here :)
I'd suggest adding new type to DriverErrorType (with same value as OdspErrorType) in minor release (main), assuming it's fine for you, and openning a bug to merge them in the future (next branch). Later is considered breaking change, so should not be done in main.

Reply:
separate draft PR made to add tokenFetchError to DrvierErrorType. https://github.com/microsoft/FluidFramework/pull/13893